docs/architecture/architecture-overview.adoc

```diff
@@ -0,0 +1,260 @@
+= Mobile Core (MC) Architecture Overview
+:toc:
+
+.Target Audience
+****
+The target audience for this document is developers who want a better understanding of what Mobile Core does, what value it brings and if they were to build a Mobile Service, how would they get MC to discover it and integrate with it.
+****
+
+== Terminology
```

Comment:
what is 'mobile core'?

Reply:
Added a definition to the Terminology section